lib/site_prism/dsl.rb

```diff
@@ -36,6 +36,14 @@ def raise_if_block(obj, name, has_block, type)
       raise SitePrism::UnsupportedBlockError
     end
 
+    # Prevent users from naming the elements starting with no_
```

Comment:
For now can this be changed to `"WARN" users if they use names which could cause DSL collisions`